idseq_dag/engine/pipeline_step.py

```diff
@@ -85,6 +85,18 @@ def validate(self):
             done_file = self.done_file(f)
             subprocess.check_call("date > %s" % done_file, shell=True)
 
+    def wait_until_finished(self):
+        while True:
+            if self.finished:
+                return
+            time.sleep(5)
```

Comment:
I always wonder if an exception should be raised in case we can prove this would never finish.   Or is there a guarantee that the flag would be set eventually?

Reply:
yeah we are missing something here. need to raise exception when run_thread fails.